Refactor/BC-345-refactor-signup-register-user-info-css-to-scss

jasonmar08 opened the pull request:
Refactored CSS for `RegisterUserInfo` and `SignUp` screens to SCSS format, utilizing mixins and variables to minimize repeated code.

HectorIlarraza commented on src/screens/RegisterUserInfo/dist/RegisterUserInfo.css:
```diff
@@ -0,0 +1,99 @@
+/* --------------------------- */
```
If I'm not mistaken this is the original CSS file you used as a template to create the SCSS file above correct? If so make sure to remove this file for any future SCSS refactor tickets to avoid possible confusion for reviewers I almost thought you missed a file lol.

HectorIlarraza commented on src/screens/SignUp/SignUp.scss:
```diff
@@ -1,3 +1,32 @@
+$color-input: rgb(248, 248, 248);
+
+// Mixins
+@mixin input-fields($padding, $width) {
```
chef's kiss 👨🏽‍🍳 🤌🏾 right here @mixin & @include

HectorIlarraza commented on src/screens/SignUp/dist/SignUp.css:
```diff
@@ -0,0 +1,106 @@
+/* --------------------------- */
```
Same for this file to be removed

HectorIlarraza reviewed (changes requested):
Fantastic SCSS usage for a first-time application @jasonmar08 just removed those CSS files and I can move it to the next phase.

jasonmar08 commented:
- Removed sass compiler extension and dist folder created by it
- Also fixed bug related to RegisterUserInfo screen where it did not updated the isSuccess uiStatus to `false` after form submission

HectorIlarraza reviewed (approved):
Everything is now perfect Jason, approved for the next phase.

yeezick commented on src/screens/RegisterUserInfo/RegisterUserInfo.scss:
```diff
@@ -1,3 +1,23 @@
+// Variables
+$color-btn: rgb(202, 207, 212);
+$color-btn-active: rgb(54, 91, 123);
+$color-btn-hover: rgb(79, 127, 169);
```
above and beyond with variables and mixins! I actually haven't learned about them yet and would love to learn from you

yeezick commented on src/screens/RegisterUserInfo/RegisterUserInfo.scss:
```diff
@@ -1,3 +1,23 @@
+  font-size: 15px;
+  height: 1.5rem;
+  margin-top: 0.4rem;
+  padding: 0.3rem 0.4rem;
```
let's resort to `px` for now to stay consistent and in control. rem, em, percentages, etc are relative and can provoke styling defects

yeezick commented on src/screens/SignUp/SignUp.scss:
```diff
@@ -1,3 +1,32 @@
+// Variables
+$color-btn: rgb(202, 207, 212);
+$color-btn-active: rgb(54, 91, 123);
```
i noticed some of these variables are repeated, see if you can migrate them to a shared scss sheet maybe in a new `styles/` directory. This may be the beginning of our component library

jasonmar08 replied:
Yeah, I was thinking the same thing and preferred to do that, but wasn't sure if we were ready to create a new main styles directory

yeezick replied:
@jasonmar08 trust ur gut, i trust ur code. anything questionable can be caught here and if i'm not reviewing don't hesitate to ping me

yeezick reviewed (changes requested):
Awesome SCSS implementation! Well formatted and readable. Added minor thoughts here but functionally great work.

jasonmar08 commented:
- Created a global scss directory which would contain global styling variables, mixins, functions, etc.

yeezick reviewed (approved):
Excellent work Jason.
